src/fmu/settings/_resources/changelog_manager.py

```diff
@@ -25,3 +31,51 @@ def __init__(self: Self, fmu_dir: FMUDirectoryBase) -> None:
     def relative_path(self: Self) -> Path:
         """Returns the relative path to the log file."""
         return Path("logs") / LogFileName.changelog
+
+    def log_updates_to_changelog(
+        self: Self,
+        updates: dict[str, Any],
+        old_resource_dict: dict[str, Any],
+        file: str,
+    ) -> None:
+        """Logs the updates to a resource to the changelog."""
+        for key, new_value in updates.items():
+            change_type = ChangeType.update
+            if "." in key:
+                old_value = self._get_dot_notation_key(
+                    resource_dict=old_resource_dict, key=key
+                )
+            else:
+                old_value = old_resource_dict.get(key)
+            if old_value is not None:
+                if isinstance(old_value, BaseModel) and isinstance(
+                    new_value, BaseModel
+                ):
+                    old_value_string = str(old_value.model_dump())
+                    new_value_string = str(new_value.model_dump())
+                else:
+                    old_value_string = str(old_value)
+                    new_value_string = str(new_value)
+                change_string = (
+                    f"Updated field '{key}'. Old value: {old_value_string}"
+                    f" -> New value: {new_value_string}"
+                )
+            else:
+                change_type = ChangeType.add
+                if isinstance(new_value, BaseModel):
+                    new_value_string = str(new_value.model_dump())
+                else:
+                    new_value_string = str(new_value)
+                change_string = f"Added field '{key}'. New value: {new_value_string}"
+
+            change_entry = ChangeInfo(
+                timestamp=datetime.now(UTC),
+                change_type=change_type,
+                user=os.getenv("USER", "unknown"),
+                path=self.fmu_dir.path,
+                change=change_string,
+                hostname=socket.gethostname(),
+                file=file,
```

Comment:
And would either have to update this model to accept a `Path` (which serializes to a string anyway), or call `str()` on it